extern/f5totec/f5totec.cpp

```diff
@@ -226,31 +228,19 @@ int main( int argc, char * argv[] ){
     tec_init = 1;
     delete [] all_vars;
 
-    // Count up the number of times each node is referred to
-    // in the discontinuous element-wise data
-    float *counts = new float[ num_points ];
-    memset(counts, 0, num_points*sizeof(float));
-    for ( int j = 0; j < ptr[num_elements]; j++ ){
-      counts[conn[j]] += 1.0;
-    }
-    for ( int i = 0; i < num_points; i++ ){
-      if (counts[i] != 0.0){
-        counts[i] = 1.0/counts[i];
-      }
-    }
-
-    // For each component, average the nodal data
-    float *avg_edata = new float[ num_points * num_evariables ];
-    // Nodally average the data
-    memset(avg_edata, 0, num_points * num_evariables*sizeof(float));
-    for ( int j = 0; j < num_evariables; j++ ){
-      for ( int k = 0; k < ptr[num_elements]; k++ ){
-        avg_edata[num_evariables*conn[k] + j] += counts[conn[k]]*edata[num_evariables*k + j];
+    // For each element, average the nodal data
+    float *avg_edata = new float[ num_elements * num_evariables ];
+    memset(avg_edata, 0, num_elements * num_evariables*sizeof(float));
+    for ( int i = 0; i < num_elements; i++ ){
+      int nnodes = ptr[i+1] - ptr[i];
+      for ( int j = 0; j < num_evariables; j++ ){
+        for ( int k = ptr[i]; k < ptr[i+1]; k++ ){
+          avg_edata[num_evariables*i + j] += edata[num_evariables*k + j];
+        }
+        avg_edata[num_evariables*i + j] /= nnodes;
```

Comment:
So we're not averaging the nodal values anymore and tecio can handle the discontinuous nodal values? Nice.